### PR TITLE
Readme documentation

### PR DESCRIPTION
**Pull request by rafaelcgs10**
Hi,

in this PR we greatly improve our README file with more documentation about our operator.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,79 +1,138 @@
...
+Furthermore, the downscale step is calculated using the amount of current nodes running and the CPU target. For example, if there are two nodes running and the CPU target is 50%, in order to downscale
+occur the CPU utilization must go bellow 25%. This is important to avoid downscale that immediately causes upscale.
+
+All scale operations are made respecting a reaction time window, which at time is not part of the manifest specification.
+
+The image bellow shows how peaks above the CPU target of 50% are shortened by the automatic increase of nodes.
+![Bigtable CPU utilization and nodes count](cpu_scaling.png "Autoscaling on CPU utilization.")
+
+## Prerequisites
+1. Enable APIs [Bigtable](https://cloud.google.com/bigtable/docs/access-control) and [Monitoring](https://cloud.google.com/monitoring/api/enable-api) on your GCP project.
```
Comment (bugarela) [inline README.md]:
```suggestion
1. Enable [Bigtable](https://cloud.google.com/bigtable/docs/access-control) and [Monitoring](https://cloud.google.com/monitoring/api/enable-api) APIs on your GCP project.
```

File: README.md [hunk 2]
```diff
@@ -1,79 +1,138 @@
...
-#### Check pods and logs
-```sh
-kubectl -n bigtable-autoscaler-system logs $(kubectl -n bigtable-autoscaler-system get pods | tail -n1 | cut -d ' ' -f1) --all-containers
-```
+# Bigtable Autoscaler Operator 
+**Bigtable Autoscaler Operator** is a [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) to autoscale the number of nodes of a [Google Cloud Bigtable](https://cloud.google.com/bigtable) instance based on the CPU utilization.
+
+## Overview
+[Google Cloud Bigtable](https://cloud.google.com/bigtable) is designed to scale horizontally, meaning that the number of nodes of an instance can be increased to balance and reduce the average CPU utilization.
+For Bigtable applications dealing with high variances of workload, the procedure of managing the amount of nodes must be fully automated for a better resource cost.
```
Comment (jeangnc) [inline README.md]:
```suggestion
For Bigtable applications dealing with high variances of workload, automating the cluster scaling allow handling short load bursts while keeping costs as low as possible.
```

File: README.md [hunk 3]
```diff
@@ -1,79 +1,138 @@
...
+
+## Development environment
+These are the steps for setting up the development environment. 
+
+This project is using go version 1.13 and other tools with its respective version, we don't guarantee that using other versions can perform successful builds.
+
+1. Install [kubebuilder](https://book.kubebuilder.io/quick-start.html#installation) version 2.3.2. 
+    1. Also make sure that you have its dependencies installed: [controller-gen](https://github.com/kubernetes-sigs/controller-tools/) version 0.5.0 and [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/) version 3.10.0 
+
+1. Follow [Option 1](#option-1) or [Option 2](#option-2) section.
```
Comment (bugarela) [inline README.md:70]:
```suggestion
1. Follow [Option 1](#option-1) or [Option 2](#option-2) section.
1. Create your secret following [Secret Setup](#secret-setup) section.
```

**Review by bugarela — COMMENTED** [review]
We could use a Table of Contents. There are some tools to generate it such as [this online one](https://ecotrust-canada.github.io/markdown-toc/).

File: README.md (hunk 3 above)
Comment (rafaelcgs10) [reply README.md:70]: Actually, this secrete-setup section also says to you to create a role binding, but I think this not needed here. Right?

Comment (bugarela) [reply README.md:70]: This are dev instructions right? We have to create the role binding at dev.

Comment (rafaelcgs10) [reply README.md:70]: Ops, yes, but I'm also referring this section in the correct step for each option.

File: README.md [hunk 4]
```diff
@@ -1,79 +1,138 @@
...
-```sh
-kubectl -n bigtable-autoscaler-system logs $(kubectl -n bigtable-autoscaler-system get pods | tail -n1 | cut -d ' ' -f1) --all-containers
-```
+# Bigtable Autoscaler Operator 
+**Bigtable Autoscaler Operator** is a [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) to autoscale the number of nodes of a [Google Cloud Bigtable](https://cloud.google.com/bigtable) instance based on the CPU utilization.
+
+## Overview
+[Google Cloud Bigtable](https://cloud.google.com/bigtable) is designed to scale horizontally, meaning that the number of nodes of an instance can be increased to balance and reduce the average CPU utilization.
+For Bigtable applications dealing with high variances of workload, the procedure of managing the amount of nodes must be fully automated for a better resource cost.
+This operator automates Bigtable scaling by reconciling the number of nodes and the CPU utilization with the `minNodes`, `maxNodes`, and `targetCPUUtilization` manifest specifications.
```
Comment (jeangnc) [inline README.md]:
```suggestion
This operator automates the scaling by balancing the number of nodes to keep the CPU utilization under the manifest specifications.
```

File: README.md [hunk 5]
```diff
@@ -1,79 +1,149 @@
...
+    + [Option 2: Manual run](#option-2--manual-run)
+  * [Secret setup](#secret-setup)
+  * [Running tests](#running-tests)
+
+## Overview
+[Google Cloud Bigtable](https://cloud.google.com/bigtable) is designed to scale horizontally, meaning that the number of nodes of an instance can be increased to balance and reduce the average CPU utilization.
+For Bigtable applications dealing with high variances of workload, the procedure of managing the amount of nodes must be fully automated for a better resource cost.
+This operator automates Bigtable scaling by reconciling the number of nodes and the CPU utilization with the `minNodes`, `maxNodes`, and `targetCPUUtilization` manifest specifications.
+
+The reconciler behavior is to keep the CPU utilization of the instance bellow the target specification, it does that while respecting the minimum and maximum amount of nodes.
```
Comment (jeangnc) [inline README.md]:
```suggestion
The reconciler's responsibility is to keep the CPU utilization of the instance below the target specification respecting the minimum and maximum amount of nodes.
```

**Review by jeangnc — CHANGES_REQUESTED** [review]

**Review by jeangnc — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (bugarela) [reply README.md:70]: whops, that's correct. Sorry!

**Review by bugarela — APPROVED** [review]

**Pull request merged**